presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java

```diff
@@ -354,9 +354,9 @@ public void testAggregationPushdown()
         //distinct
         assertThat(query("SELECT distinct shippriority,clerk FROM orders")).isCorrectlyPushedDown();
 
-        assertThat(query("SELECT approx_distinct(custkey) FROM orders")).isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
-        assertThat(query("SELECT approx_distinct(totalprice) FROM orders")).isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
-        assertThat(query("SELECT approx_distinct(comment) FROM orders")).isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
-        assertThat(query("SELECT approx_distinct(__time) FROM orders")).isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
+        assertThat(query("SELECT approx_distinct(custkey) FROM orders")).ordered().isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
+        assertThat(query("SELECT approx_distinct(totalprice) FROM orders")).ordered().isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
+        assertThat(query("SELECT approx_distinct(comment) FROM orders")).ordered().isCorrectlyPushedDown(ImmutableList.of(Optional.of(100L)));
```

Comment:
This works here only because this is a global aggregation and there is exactly one row.
There is no ordering otherwise.
Using `ordered` here is just a hack to get tolerances to work.

instead you could impl tolerances for single-row results in https://github.com/prestosql/presto/blob/6c6fa9cc26c796efb581f7fb39f9790f8e641eb1/presto-main/src/test/java/io/prestosql/sql/query/QueryAssertions.java#L306

Reply:
I wanted to make this generic rather then only make it work for the single value for a single column. Right now I have updated the code so it takes a list of tolerance percentages and applies that for all the values of a given column. Let me know if you disagree with that approach.